File: Deepseek-Coder-33b-Instruct/stage2/acc_unmap_data.cpp
#include <stdlib.h>
#include <openacc.h>

#define N 100

int main() {
    int *host_array = (int*)malloc(N * sizeof(int));
    int *device_ptr;

    // Initialize host array
    for(int i = 0; i < N; i++) {
        host_array[i] = i;
    }

    // Map host array to device
    #pragma acc data copyin(host_array[0:N])
    {
        device_ptr = host_array;
    }

    // Unmap host array from device
    acc_unmap_data(host_array);

    // Check if data is still accessible
    for(int i = 0; i < N; i++) {
        if(host_array[i] != i) {
            printf("Data is not accessible after unmap\n");
            return 1;
        }
    }

    printf("Data is accessible after unmap\n");
    return 0;
}

File: Deepseek-Coder-33b-Instruct/stage2/acc_unmap_data.f90
#ifndef T1
!T1:acc unmap data,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(KIND=4), DIMENSION(:), ALLOCATABLE :: a
        REAL(KIND=4), DIMENSION(:), POINTER :: a_d
        INTEGER :: i, n

        n = 100
        ALLOCATE(a(1:n))
        ALLOCATE(a_d(1:n))

        ! Initialize host data
        DO i = 1, n
          a(i) = REAL(i)
        END DO

        ! Map data to device
        a_d => a
        CALL acc_map_data(a_d, n)

        ! Unmap data from device
        CALL acc_unmap_data(a_d)

        ! Check if unmap was successful
        IF (ASSOCIATED(a_d)) THEN
          errors = errors + 1
        END IF

        DEALLOCATE(a)
        DEALLOCATE(a_d)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM